File: src/main/java/no/nav/sosialhjelp/soknad/Application.java
package no.nav.sosialhjelp.soknad;

import no.nav.sosialhjelp.soknad.domain.model.mock.MockUtils;
import no.nav.sosialhjelp.soknad.domain.model.util.ServiceUtils;
import org.springframework.boot.Banner;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;

@SpringBootApplication
public class Application extends SpringBootServletInitializer {

    @Override
    protected SpringApplicationBuilder configure(SpringApplicationBuilder builder) {
        return configureApplication(builder);
    }

    public static void main(String[] args) {
        configureApplication(new SpringApplicationBuilder()).run(args);
    }

    private static SpringApplicationBuilder configureApplication(SpringApplicationBuilder builder) {

        if (!ServiceUtils.envIsNonProduction() && MockUtils.isMockAltProfil()) {
            throw new Error("mockAltProfil har blitt satt til true i prod. Stopper applikasjonen da dette er en sikkerhetsrisiko.");
        }

        if (!ServiceUtils.envIsNonProduction() && MockUtils.isRunningWithInMemoryDb()) {
            throw new Error("no.nav.sosialhjelp.soknad.hsqldb har blitt satt til true i prod. Stopper applikasjonen da dette er en sikkerhetsrisiko.");
        }

        if (!ServiceUtils.envIsNonProduction() && (MockUtils.isAlltidHentKommuneInfoFraNavTestkommune() || MockUtils.isAlltidSendTilNavTestkommune())) {
            throw new Error("Alltid send eller hent fra NavTestkommune er satt til true i prod. Stopper applikasjonen da dette er en sikkerhetsrisiko.");
        }

        return builder
                .sources(Application.class)
                .bannerMode(Banner.Mode.OFF);
    }

}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/Application.kt
package no.nav.sosialhjelp.soknad

import no.nav.sosialhjelp.soknad.app.MiljoUtils
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.builder.SpringApplicationBuilder
import java.time.LocalDateTime
import java.time.ZoneId
import java.time.ZonedDateTime
import java.time.temporal.ChronoUnit

@SpringBootApplication
class Application

fun main(args: Array<String>) {
    configureApplication(SpringApplicationBuilder())
        .run(*args)
        .registerShutdownHook()
}

private fun configureApplication(builder: SpringApplicationBuilder): SpringApplicationBuilder {
    if (!MiljoUtils.isNonProduction() && MiljoUtils.isMockAltProfil()) {
        throw Error("Spring profile `mock-alt` har blitt satt i prod. Stopper applikasjonen da dette er en sikkerhetsrisiko.")
    }
    return builder
        .sources(Application::class.java)
}

fun nowWithMillis(): LocalDateTime =
    ZonedDateTime.now(ZoneId.of("Europe/Oslo")).toLocalDateTime().truncatedTo(ChronoUnit.MILLIS)
